src/object/notification.c

```diff
@@ -187,6 +184,10 @@ void receiveSignal(tcb_t *thread, cap_t cap, bool_t isBlocking)
 
             notification_ptr_set_state(ntfnPtr, NtfnState_Waiting);
             ntfn_ptr_set_queue(ntfnPtr, ntfn_queue);
+
+#ifdef CONFIG_KERNEL_MCS
+            maybeReturnSchedContext(ntfnPtr, thread);
```

Comment:
This should alter the behaviour of  the call to `scheduleTCB`? I don't see how that is ok.

Reply:
Regardless of whether `scheduleTCB` is called before or after `maybeReturnSchedContext`, the call to `scheduleTCB` within `maybeReturnSchedContext` will always return false as the thread is always `ThreadState_BlockedOnNotification` at that point. As such, this particular call to `scheduleTCB` doesn't depend on whether the thread in question has a bound SC or the state of any bound SC.